go/obscuronode/enclave/attestation.go

```diff
@@ -0,0 +1,75 @@
+package enclave
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"github.com/obscuronet/obscuro-playground/go/log"
+
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+type IdData struct {
+	Owner  common.Address
+	PubKey []byte
+}
+
+type AttestationProvider interface {
+	// GetReport returns the verifiable attestation report
+	GetReport(pubKey []byte, owner common.Address) (obscurocommon.AttestationReport, error)
+	// VerifyReport returns the embedded report data
+	VerifyReport(att obscurocommon.AttestationReport) ([]byte, error)
+}
+
+type EgoAttestationProvider struct{}
+
+func (e *EgoAttestationProvider) GetReport(pubKey []byte, owner common.Address) (obscurocommon.AttestationReport, error) {
+	idHash := getIdHash(owner, pubKey)
+	report, err := enclave.GetRemoteReport(idHash)
+	if err != nil {
+		return obscurocommon.AttestationReport{}, err
+	}
+
+	return obscurocommon.AttestationReport{
+		Report: report,
+		PubKey: pubKey,
+		Owner:  owner,
+	}, nil
+}
+
+// todo: we should probably return other properties for manual verification, not just the data (e.g. validate code hash)
+func (e *EgoAttestationProvider) VerifyReport(att obscurocommon.AttestationReport) ([]byte, error) {
+	remoteReport, err := enclave.VerifyRemoteReport(att.Report)
+	if err != nil {
+		log.Log(fmt.Sprintf("Attestation failed: %s", err))
+		return []byte{}, err
+	}
+	return remoteReport.Data, nil
+}
+
+type MockAttestationProvider struct{}
+
+func (e *MockAttestationProvider) GetReport(pubKey []byte, owner common.Address) (obscurocommon.AttestationReport, error) {
```

Comment:
Kinda nitpicky, so please bear with me :)

Can this be changed to HardcodedAttestationProvider or StaticAttestationProvider, or maybe there's a better name ? I feel like Mock is something a bit different, usually wire-frames for mocking services in tests.

Reply:
Yeah fair point, I've gone for DummyAttestationProvider to be consistent with Joel's RPC server and because I want it to be clear that it provides no useful value at all whereas Hardcoded and Static sound like they could be legit implementations.

Reply:
Sounds good 👍